src/main/java/reposense/report/RepoCloner.java

```diff
@@ -57,23 +56,28 @@ public RepoConfiguration getClonedRepo(String outputPath) throws IOException {
             return null;
         }
 
-        if (isPreviousRepoDifferent()) {
-            currentRepoDefaultBranch = GitBranch.getCurrentBranch(configs[currentIndex].getRepoRoot());
-        }
-        configs[currentIndex].updateBranch(currentRepoDefaultBranch);
-
+        currentRepoDefaultBranch = GitBranch.getCurrentBranch(configs[currentIndex].getRepoRoot());
         cleanupPrevRepoFolder();
 
+        previousIndex = currentIndex;
+        currentIndex = (currentIndex + 1) % configs.length;
+        return configs[previousIndex].getLocation();
+    }
+
+    /**
+     * Prepares repo specified by {@code config} to be analyzed by updating and checking out its branch.
+     */
+    public boolean prepareToAnalyze(String outputPath, RepoConfiguration config) throws IOException {
```

Comment:
doesn't seems to explain what does the boolean return type stand for, was it even used?

Reply:
In this method, the branch of the config is checked out and the method returns false if it fails. This is so that ReportGenerator knows to skip the repo. Maybe throwing exception is the cleaner option.